core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/codesystem/CodeSystemUpgradeRequest.java

```diff
@@ -0,0 +1,139 @@
+
+/**
+ * @since 7.13.0 
+ */
+final class CodeSystemUpgradeRequest implements Request<RepositoryContext, String> {
+
+	private static final long serialVersionUID = 1L;
+
+	@NotNull
+	private final CodeSystemURI codeSystem;
+	
+	@NotNull
+	private final CodeSystemURI extensionOf;
+
+	private String codeSystemId;
+	
+	public CodeSystemUpgradeRequest(CodeSystemURI codeSystem, CodeSystemURI extensionOf) {
+		this.codeSystem = codeSystem;
+		this.extensionOf = extensionOf;
+	}
+	
+	void setCodeSystemId(String codeSystemId) {
+		this.codeSystemId = codeSystemId;
+	}
+	
+	@Override
+	public String execute(RepositoryContext context) {
+		if (!codeSystem.isHead()) {
+			throw new BadRequestException("Upgrades can not be started from CodeSystem versions.")
+				.withDeveloperMessage("Use '%s' only instead of '%s'", codeSystem.getCodeSystem(), codeSystem);
+		}
+
+		// get available upgrades 
+		final CodeSystem currentCodeSystem = CodeSystemRequests.prepareGetCodeSystem(codeSystem.getCodeSystem())
+				.setExpand(CodeSystem.Expand.AVAILABLE_UPGRADES + "()")
+				.build()
+				.execute(context);
+		final List<CodeSystemURI> availableUpgrades = currentCodeSystem.getAvailableUpgrades();
+		// report bad request if there are no upgrades available
+		if (availableUpgrades.isEmpty()) {
+			throw new BadRequestException("There are no upgrades available for the CodeSystem '%s'.", codeSystem.getCodeSystem());
+		}
+		
+		// or the selected extensionOf version is not present as a valid available upgrade
+		if (!availableUpgrades.contains(extensionOf)) {
+			throw new BadRequestException("Upgrades can only be performed to the next available version dependency.")
+				.withDeveloperMessage("Use '%s/<VERSION_ID>', where <VERSION_ID> is one of: '%s'", extensionOf.getCodeSystem(), availableUpgrades);
+		}
+		
+		// auto-generate the codeSystemId if not provided
+		// auto-generated upgrade IDs consist of the original CodeSystem's name and the new extensionOf dependency's URI
+		final String upgradeCodeSystemId;
+		if (codeSystemId == null) {
+			upgradeCodeSystemId = String.format("%s-%s-UPGRADE", codeSystem.getCodeSystem(), extensionOf.getPath() /*versionId*/); 
+		} else if (codeSystemId.isBlank()) {
+			throw new BadRequestException("'codeSystemId' property should not be empty, if provided");
+		} else {
+			upgradeCodeSystemId = codeSystemId;
+		}
+		
+		// create the same branch name under the new extensionOf path
+		String parentBranch = context.service(ResourceURIPathResolver.class).resolve(context, List.of(extensionOf)).stream().findFirst().get();
+		
+		String upgradeBranch = RepositoryRequests.branching().prepareCreate()
+			.setParent(parentBranch)
+			.setName(codeSystem.getCodeSystem())
+			.build()
+			.execute(context);
+		
+		// merge branch content from the current code system to the new upgradeBranch
+		Merge merge = RepositoryRequests.merging().prepareCreate()
+			.setSource(currentCodeSystem.getBranchPath())
+			.setTarget(parentBranch)
```

Comment:
Didn't this change to `upgradeBranch` in a commit? I could swear I saw it earlier 😕

Reply:
It did change to that, it looks like that in my dev env.